File: scripts/extract-pdf.mjs
import fs from 'node:fs/promises'
import path from 'node:path'
import * as pdfjsLib from 'pdfjs-dist/legacy/build/pdf.mjs'

const inputPath = path.resolve('public/docs/reidcollins.pdf')
const outputPath = path.resolve('public/docs/reidcollins.txt')

async function extractTextFromPage(pdf, pageNum) {
  const page = await pdf.getPage(pageNum)
  const content = await page.getTextContent()
  const strings = content.items.map((item) => ('str' in item ? item.str : ''))
  return strings.join(' ').replace(/\s+/g, ' ').trim()
}

async function main() {
  try {
    const data = await fs.readFile(inputPath)
    const uint8 = new Uint8Array(data.buffer, data.byteOffset, data.byteLength)
    const loadingTask = pdfjsLib.getDocument({ data: uint8, useWorker: false })
    const pdf = await loadingTask.promise
    const pages = pdf.numPages
    const parts = []
    for (let i = 1; i <= pages; i += 1) {
      const text = await extractTextFromPage(pdf, i)
      parts.push(text)
    }
    const finalText = parts.join('\n\n--- PAGE BREAK ---\n\n')
    await fs.writeFile(outputPath, finalText, 'utf8')
    console.log(`Extracted text -> ${outputPath}`)
    console.log(`Pages: ${pages}`)
  } catch (err) {
    console.error('Failed to extract PDF:', err)
    process.exitCode = 1
  }
}

main()

File: scripts/extract_pdf.py
#!/usr/bin/env python3
"""Extract plain text from the resume PDF.

Replaces the Node-based extract-pdf.mjs script.
Uses pdfplumber for consistent text extraction across all Python scripts.

Usage:
    python scripts/extract_pdf.py
"""

import sys
from pathlib import Path

try:
    import pdfplumber
except ImportError:
    sys.exit("pdfplumber is required.  Install it with:\n  pip install pdfplumber")

ROOT = Path(__file__).resolve().parent.parent
INPUT_PATH = ROOT / "public" / "docs" / "reidcollins.pdf"
OUTPUT_PATH = ROOT / "public" / "docs" / "reidcollins.txt"


def main():
    if not INPUT_PATH.exists():
        sys.exit(f"PDF not found: {INPUT_PATH}")

    pages: list[str] = []
    with pdfplumber.open(str(INPUT_PATH)) as pdf:
        for page in pdf.pages:
            text = page.extract_text()
            if text:
                pages.append(text)

    final_text = "\n\n--- PAGE BREAK ---\n\n".join(pages)
    OUTPUT_PATH.write_text(final_text, encoding="utf-8")
    print(f"Extracted text -> {OUTPUT_PATH}")
    print(f"Pages: {len(pages)}")


if __name__ == "__main__":
    main()
